Lib/asyncio/futures.py

```diff
@@ -273,8 +273,13 @@ def set_exception(self, exception):
         if isinstance(exception, type):
             exception = exception()
         if type(exception) is StopIteration:
```

Comment:
While we're at it, let's change this to `isinstance(exception, StopIteration)` -- subclasses of `StopIteration` are just as bad as instances of the base class. (And do this in the C version as well.)